README.md

```diff
@@ -177,44 +177,65 @@ Example of a partial CR:
+Where:
+* `<mountPathPrefix>` defaults to `/platform/bindings` if not specified in the `ServiceBinding` CR via the `mountPathPrefix` element.  The environment variable `SERVICE_BINDINGS_ROOT` **MUST** be set to the value of `<mountPathPrefix>`, so that applications can always find this directory, and it is immutable.  This means that if another `ServiceBinding` CR wants to project itself into the same container, it **MUST** reuse the current `SERVICE_BINDINGS_ROOT` value.  
+* `<service-id>` is the `id` field of the corresponding `service` entry in the `ServiceBinding` CR.  If the `id` field is not present, the `name` field is used instead.  The `<service-id>` path **MUST** be unique between the services bound to a particular application.
+* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+
+Example:  `/platform/bindings/mongo-db/secret/MONGODB_HOST`
+
+#### Cross services data mappings
+
+The `ServiceBinding` CR allows for the declaration of mappings that are potentially composed of different service bindings (ids).  For this case, the special id of `dataMappings` is reserved and **MUST NOT** be used by any service bindings.  
+
+This means that a container **MAY** have the path `<mountPathPrefix>/dataMappings/secret/<persisted_secret>` mounted, representing all of the composed bindings.  
+
+#### Exposing data as environment variables
 
-The decision to mount vs inject is made in the following ascending order of precedence:
+The specification allows for binding properties to be additionally exposed as environment variables.
+
+The decision to make it available as an environment variable is made in the following ascending order of precedence:
 * value of the `bindAs` attribute in the backing service as defined in its [annotations](annotations.md#data-model--building-blocks-for-expressing-binding-information), applying to the binding item referenced by the annotation.
 * value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
 * value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
 
-#### Injecting data
-
-The key `SERVICE_BINDINGS` acts as a global map of the service bindings and **MUST** always be injected into the environment.  It contains a JSON payload with `bindingKeys` key name containing a list of all available binding information available. Each item of the `bindingKeys` list includes an object containing `name`, `bindAs` and an optional `mountPath` (if it is bound as a volume).
-
-Example:
-
-```json
-SERVICE_BINDINGS = {
-  "bindingKeys": [
-    {
-      "name": "KAFKA_USERNAME",
-      "bindAs": "envVar"
-    },
-    {
-      "name": "KAFKA_PASSWORD",
-      "bindAs": "volume",
-      "mountPath": "/platform/bindings/secret/"
-    }
-  ]
-}
-```
+If a particular binding key is made available as an environment variable an entry **MUST** be made at the following file:
```

Comment:
The intent is to enable both scenarios:
(1) when the entity that is projecting the bindings is also the one that is making the env var available
(2) when some runtime library is walking over the mounted bindings and creating env vars for some of the bindings (presumably during app startup)

In both cases the env vars are available for the application to consume, and there's a consistent way that describes which env vars are available (agnostic to how things were projected or processed).

Reply:
I'm very wary of standardizing how the canonical representation is transformed into environment variables or even whether it is.  At the very least, I don't think this should be a part of the core spec, but rather an extension specification defined as "this is optional functionality that not all platforms need to support, but if they do this is how it should be done".

Reply:
In general, I consider this functionality a convenience and conventions should emerge over time as the community decides how they like describing this functionality rather than prescribing it up front.  For example, in many of the Cloud Native Buildpacks that currently exist, we wouldn't use this file as [we describe a convention](https://github.com/paketo-buildpacks/azure-application-insights/blob/master/properties/properties.go#L44-L48) like the following:

```
for k, v := range b.Secret {
  s := strings.ToUpper(k)
  s = strings.ReplaceAll(s, "-", "_")
  s = strings.ReplaceAll(s, ".", "_")

  vars = append(vars, fmt.Sprintf(`export APPINSIGHTS_%s="%s"`, s, v))
}
```

This convention-based approach means that we can map *every* secret, regardless of whether it's been defined in a JSON-structure to an environment variable giving flexibility to the provider (especially in the case where that provider is a user manually generating the binding)